File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        vector<pair<int, int>> v;

        for(auto it:nums){
            mp[it]++;
        }

        for(auto it:mp){
            v.push_back({it.second, it.first});
        }
        	sort(v.begin(), v.end(), greater<pair<int,int>>());


        vector<int> ans;

        int i=0;
        while(i<k && i<v.size()){
            ans.push_back(v[i].second);
            i++;
        }

        return ans;
        
    }
};

File: Data Structures & Algorithms/reverse-bits/submission-0.java
class Solution {
    public int reverseBits(int n) {
        int ans = 0;
        int cnt = 32;
        while (n > 0) {
            int t = (n & 1);
            n = n >> 1;

            ans = ans << 1;
            ans = ans | t;
            cnt--;
        }
        ans = ans << cnt;

        return ans;
    }
}
